### PR TITLE
Implementation of OpenTelemetry metrics transport

### PR DESCRIPTION
**Pull request by h1k3r**
# OpenTelemetry transport implementation

This pull request introduces several enhancements and configurations to support metrics collection and reporting within the OpenTelemetry repository.  

## Changes

1. Changed docker setup to support metrics:  
- Added volume for Grafana data.
- Configured otel-collector to listen on all hosts.
- Added debugging exporter.
- Added Prometheus exporter.
- Configured metrics workflow.
- Configured Prometheus to collect data from otel-collector.
- Changed Tempo volume mapping for updated path in the new image version.
- Added Datadog Export to otel-collector Configuration:  
- Configured Datadog agent.
- Added forwarding of metrics to Datadog agent (commented).
- Added forwarding of metrics directly to Datadog API.

2. Added OpenTelemetry Transport for Metrics:
- Implemented a new transport for handling metrics using OpenTelemetry.
- Added basic configuration to make possible enabling/disabling the transport and setting the metrics namespace.

3. Added OpenTelemetry Metric Tests:
- Added phpunit as a dependency.
- Configured two test suites: unit and integration.
- Added basic tests for metrics transport.

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			count: 1
+			path: src/Instrumentation/CommandInstrumentation.php
+
+		-
+			message: "#^Access to constant STATUS_CODE_ERROR on an unknown class Opentelemetry\\\\Proto\\\\Trace\\\\V1\\\\Status\\\\StatusCode\\.$#"
+			count: 1
+			path: src/Instrumentation/CommandInstrumentation.php
+
+		-
+			message: "#^Method Shopware\\\\OpenTelemetry\\\\Instrumentation\\\\CommandInstrumentation\\:\\:register\\(\\) has no return type specified\\.$#"
```
Comment (mitelg) [inline phpstan-baseline.neon:14]: would be cool, if you could easy stuff like no type for parameter or return values

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -0,0 +1,66 @@
+parameters:
+	ignoreErrors:
+		-
+			message: "#^Access to constant SPAN_KIND_INTERNAL on an unknown class Opentelemetry\\\\Proto\\\\Trace\\\\V1\\\\Span\\\\SpanKind\\.$#"
```
Comment (mitelg) [inline phpstan-baseline.neon:4]: I wonder why PHPStan could not recognise this :thinking: isn't this part of the dependencies?

Comment (h1k3r) [reply phpstan-baseline.neon:4]: open-telemetry/sdk is not in the dependencies list, it's an optional dependency. Plugin's code for autoinstrumentation initialization checks if SDK is available
https://github.com/shopware/opentelemetry/blob/main/_register.php#L12

Newly added metrics implementation also does not require sdk, if it's absent - open-telemetry/api will just use noop meter.

Solution here could be including open-telemetry/sdk at least as dev dependency [(similar problem discussed on reddit)](https://www.reddit.com/r/PHP/)

Comment (mitelg) [reply phpstan-baseline.neon:4]: sounds like a reasonable solution :+1:

Comment (shyim) [reply phpstan-baseline.neon:4]: It is part of our requirement as we require `ext-opentelemetry`. Make phpstan does not understand the classes of that extension.

We added the if condition when maybe the build environment had that extension, but not the runtime and not to explode the whole thing

File: phpstan-baseline.neon (hunk 1 above)
Comment (h1k3r) [reply phpstan-baseline.neon]: Fixed most of them

File: phpstan-baseline.neon (hunk 2 above)
Comment (h1k3r) [reply phpstan-baseline.neon]: Fixed. Seems there was a dependency on the files that exist only in https://github.com/open-telemetry/opentelemetry-php. Replaced with the files from open-telemetry/api package.

**Review by mitelg — APPROVED** [review]

**Review by Gaitholabi — APPROVED** [review]

File: src/Metrics/Transports/OpenTelemetryMetricTransport.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                $metric->description,
+            )->record($metric->value, $attributes, $context);
+    }
+
+    /**
+     * @param Attributes $attributes
+     */
+    public function handleGauge(Gauge $metric, iterable $attributes, ?ContextInterface $context): void
+    {
+        // todo: replace implementation with sync gauge as soon as it's released in SDK
```
Comment (keulinho) [inline src/Metrics/Transports/OpenTelemetryMetricTransport.php:91]: can you link to the issue/PR or something from the SDK here? So we know where to look?

File: tests/integration/OpenTelemetryMeterFactoryTest.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?php
```
[the remark this answers is not in the document]
Comment (h1k3r) [reply tests/integration/OpenTelemetryMeterFactoryTest.php:1]: I kept it like in other files in the repository (see main branch):
https://github.com/shopware/opentelemetry/blob/main/src/Instrumentation/CommandInstrumentation.php

File: src/Metrics/Transports/OpenTelemetryMetricTransport.php (hunk 3 above)
Comment (h1k3r) [reply src/Metrics/Transports/OpenTelemetryMetricTransport.php:91]: There is no issue/pr, they have it implemented in the main branch, but it's not tagged yet

File: tests/integration/OpenTelemetryMeterFactoryTest.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (keulinho) [reply tests/integration/OpenTelemetryMeterFactoryTest.php:1]: @shyim was it by design?

[the remark this answers is not in the document]
Comment (Gaitholabi) [reply tests/integration/OpenTelemetryMeterFactoryTest.php:1]: I don't think so, probably just missing the CS rule

[the remark this answers is not in the document]
Comment (h1k3r) [reply tests/integration/OpenTelemetryMeterFactoryTest.php:1]: I will add the rule

[the remark this answers is not in the document]
Comment (h1k3r) [reply tests/integration/OpenTelemetryMeterFactoryTest.php:1]: Done

**Review by shyim — APPROVED** [review]

**Review by keulinho — APPROVED** [review]

**Pull request merged**